### PR TITLE
[MM-59825] Fix linter error found by updated govet

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
https://github.com/mattermost/mattermost-govet/pull/40 fixes a bug in `govet` that blocked most linter from running. Since that bug exists, some linter errors have crept into the code base. This PR fixes them.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-59825

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by enahum — APPROVED** [review]

**Review by devinbinnie — APPROVED** [review]

**Comment by hanzei** [comment]
Waiting for https://github.com/mattermost/mattermost-govet/pull/40 to get merged.

**Comment by cpoile** [comment]
@amyblais This is what needs to be cherry picked to 9.11

**Comment by hanzei** [comment]
@cpoile Why do we need to cherry-pick the change into a release branch? Because CI fails on the branch?

**Comment by cpoile** [comment]
@hanzei yes: https://github.com/mattermost/mattermost/pull/27741

**Comment by amyblais** [comment]
@hanzei Please help submit manual cherry-pick PRs for ``release-9.10``, ``release-9.9``, ``release-9.8`` and ``release-9.5`` branches. (some cherry-pick PRs are currently failing, e.g. https://github.com/mattermost/mattermost/pull/27769 and https://github.com/mattermost/mattermost/pull/27768).

**Comment by hanzei** [comment]
@amyblais Updating all of these branches manually is quite cumbersome, given that there are also required `enterprise` changes. Should we instead pin the linter version on each branch and avoid breaking CI that way?

**Comment by hanzei** [comment]
cc @isacikgoz (for XYZ) and @phoinixgrr (for delivery team)

**Comment by amyblais** [comment]
> Should we instead pin the linter version on each branch and avoid breaking CI that way?

@hanzei Should I open a ticket for you or for XYZ team?

**Comment by hanzei** [comment]
I would say yes, but I'm leaving the final call to @isacikgoz

**Comment by amyblais** [comment]
> I would say yes, but I'm leaving the final call to

Opened a ticket for XYZ: https://mattermost.atlassian.net/browse/MM-59940.

**Comment by agnivade** [comment]
I have mixed feelings about this. One one side, I feel the pain of doing backports multiple times. But, OTOH this was how it used to work in Gitlab and it was a massive pain and a mess of branches.

The issue with this is:
1. Now that you would need to cut release branches of mattermost-govet for every release which adds yet one more step in the process. 
2. For every release, you would need to pin to that release branch of mattermost-govet.

So even if nothing happens, these 2 things would need to be done every single time.

Ideally, the best option would be to include mattermost-govet inside the mono repo itself, so that all of these concerns will magically go away. All of these branch pinning stuff is just a workaround for this fundamental problem.

LMK your thoughts @hanzei @isacikgoz

**Comment by isacikgoz** [comment]
I was thinking pinning the mattermost-govet version on the Makefile, whenever we update it we don't necessarily need to do anything special for previous versions right? Right now we are always downloading the most recent version.

**Comment by agnivade** [comment]
Ah yes, that also works!

**Comment by amyblais** [comment]
@hanzei Would you be able to help with a manual cherry-pick PR for ``release-9.5`` branch to unblock https://github.com/mattermost/mattermost/pull/27769?

Agniva mentioned that https://mattermost.atlassian.net/browse/MM-59940 is a long-term fix https://github.com/mattermost/mattermost/pull/27769#issuecomment-2278109373.

**Pull request merged**
